src/Compatibility/GenAPI/Microsoft.DotNet.GenAPI.Task/GenAPITask.cs

```diff
@@ -62,17 +65,25 @@ public class GenAPITask : TaskBase
         /// <inheritdoc />
         protected override void ExecuteCore()
         {
-            GenAPIApp.Run(new MSBuildLog(Log),
-                Assemblies!,
-                AssemblyReferences,
-                OutputPath,
-                HeaderFile,
-                ExceptionMessage,
-                ExcludeApiFiles,
-                ExcludeAttributesFiles,
-                RespectInternals,
-                IncludeAssemblyAttributes
-            );
+            Debug.Assert(Assemblies != null, "Assemblies cannot be null.");
```

Comment:
I think you can decorate the property above with `required` which will enforce it to be non null. I'm also not sure why the `string[]?` is nullable.

Reply:
> I'm also not sure why the string[]? is nullable, that looks wrong.

Glad you mention it. I thought the same thing. I'll make it non-nullable, maybe nothing will break, just a couple of nullability adaptations will be needed and that's all.

Reply:
> I think you can add the required keyword to the property above

Interestingly, I see it's already decorated with `Microsoft.Build.Framework.RequiredAttribute`. I am not familiar with it, but I suspect it only applies to usage as a task.

Reply:
Okay, I don't think I want to make a change here. It's working fine. I'll leave it nullable.

If I make this property non-nullable, then I am forced to add the required modifier. But adding the required modifier forces me to add a default constructor that sets a default value to the property. If I add a default constructor, then I am forced to add required modifiers to the other properties.